### PR TITLE
Delete older logs from GCS

### PR DESCRIPTION
**Pull request by Rajalakshmi-Girish**
This PR adds `main.go` and `config.yml`.
Config.yml contains the directory names and the `daysLimit` for three jobs whose log folders are currently residing in GCS.
code checks for objects older than the` daysLimit` specified in the `config.yml` file for the respective jobs and deletes them.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+	"cloud.google.com/go/storage"
+	"context"
+	"flag"
+	"fmt"
+	"google.golang.org/api/iterator"
+	"google.golang.org/api/option"
+	"gopkg.in/yaml.v2"
+	"log"
+	"os"
+	"time"
```
Comment (mkumatag) [inline main.go:13]: try to create different sections for built-in library vs user-defined ones,
```suggestion
	"context"
	"flag"
	"fmt"
	"log"
	"os"
	"time"
	
	"cloud.google.com/go/storage"
	"google.golang.org/api/iterator"
	"google.golang.org/api/option"
	"gopkg.in/yaml.v2"
```

File: main.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+)
+// Config struct for Directory, Bucket and GCS credential path details
+type Config struct {
+	Dir          string `yaml:"dir"`
+	DaysLimit    int `yaml:"daysLimit"`
+}
+
+func NewConfig(configFile string) ([]Config, error) {
```
Comment (mkumatag) [inline main.go:22]: This function no need to be Public
```suggestion
func newConfig(configFile string) ([]Config, error) {
```

File: main.go [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+		fmt.Println("\nObject Name : ", attrs.Name)
+		fmt.Println("\nObject Created Time : ", attrs.Created)
+		loc, _ := time.LoadLocation("UTC")
+		now := time.Now().In(loc)
+		fmt.Println("\nToday : ", loc, " Time : ", now)
+		pastDate := attrs.Created
+		diff := now.Sub(pastDate)
+		days := int(diff.Hours() / 24)
+		fmt.Printf("Diffrence in days : %d days\n", days)
+		if days >= reqDays {
```
Comment (mkumatag) [inline main.go:87]: I see a lot variables declared here, they aren't required, you can get rid of a couple of them.

File: main.go [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+			o:= client.Bucket(bucket).Object(attrs.Name)
+			if err := o.Delete(ctx); err != nil {
```
Comment (mkumatag) [inline main.go:89]: again you don't need variable `o` here, you can merge both these lines

**Review by mkumatag — CHANGES_REQUESTED** [review]

**Comment by mkumatag** [comment]
@Rajalakshmi-Girish anything else pending in this PR?

**Comment by Rajalakshmi-Girish** [comment]
@mkumatag I think, that's it for now.

**Comment by ltccci** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/ppc64le-cloud/gcs-cleaner/pull/1#issuecomment-780439269" title="Approved">mkumatag</a>*, *<a href="https://github.com/ppc64le-cloud/gcs-cleaner/pull/1#" title="Author self-approved">Rajalakshmi-Girish</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=ppc64le-cloud%2Fgcs-cleaner).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/ppc64le-cloud/gcs-cleaner/blob/main/OWNERS)~~ [Rajalakshmi-Girish,mkumatag]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**